frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -65,9 +96,36 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
         })
         .catch(() => {});
     },
-    [close, handlePromise, k8sKill, subscription],
+    [close, deleteAllOperands, handlePromise, k8sKill, models, operands, subscription],
   );
 
+  const subscriptionName = subscription?.spec?.name;
+  const subscriptionNamespace = subscription?.metadata?.namespace;
+
+  React.useEffect(() => {
+    getOperands(subscriptionName, subscriptionNamespace)
+      .then((operandsForOperator) => {
+        if (operandsForOperator) {
+          setOperands(operandsForOperator);
+        } else {
+          setLoadOperandsErrorMessage(
+            `Error retrieving Operands for ${subscriptionName} in ${subscriptionNamespace}`,
+          );
+        }
+        setOperandsLoaded(true);
+      })
+      .catch((err) => {
+        setLoadOperandsErrorMessage(`Error loading Operands for ${subscriptionName}: ${err}`);
+        setOperandsLoaded(true);
+      });
+  }, [
+    subscriptionName,
+    subscriptionNamespace,
+    setOperandsLoaded,
+    setLoadOperandsErrorMessage,
+    models,
```

Comment:
I don't see where `models` is used in the `useEffect` hook.

Reply:
replaced by  `useOperandsWatcher`